src/mono/wasm/runtime/managed-exports.ts

```diff
@@ -22,8 +22,8 @@ export function init_managed_exports(): void {
     if (!runtimeHelpers.runtime_interop_exports_class)
         throw "Can't find " + runtimeHelpers.runtime_interop_namespace + "." + runtimeHelpers.runtime_interop_exports_classname + " class";
 
-    const install_sync_context = get_method("InstallSynchronizationContext");
-    mono_assert(install_sync_context, "Can't find InstallSynchronizationContext method");
+    const install_sync_context = cwraps.mono_wasm_assembly_find_method(runtimeHelpers.runtime_interop_exports_class, "InstallSynchronizationContext", -1);
```

Comment:
you could also add

```typescript
import MonoWasmThreads from "consts:monoWasmThreads";
```

and

```typescript
  const install_sync_context = MonoWasmThreads && wraps.mono_wasm_assembly_find_method(runtimeHelpers.runtime_interop_exports_class, "InstallSynchronizationContext", -1);
  ```
  
which I think will encourage rollup to remove the `if (install_sync_context)` below, completely.

Reply:
Do we really need to compile it out? I would prefer for it to work correctly if the .wasm file is swapped out, since that's a thing people do for testing.

Reply:
>  work correctly if the .wasm file is swapped out

that will not work to various degrees.

1. Emscripten builds `dotnet.js` pretty differently depending on the `-s` flags passed to `emcc`.  In particular it leaves out the pthreads support library.  but also it might use the single-threaded libc.
2. I uh... use the `MonoWasmThreads` to shake out all of Mono's pthread support JS code.

if we want to support swapping the .wasm files without swapping dotnet.js that would be a whole project.


---

That said, I don't strongly feel that this particular `if` statement needs to get removed.  It's fine as-is